src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py

```diff
@@ -377,11 +377,19 @@ def _update_artifacts(artifacts, lab_resource_id):
 
     result_artifacts = []
     for artifact in artifacts:
-        artifact_id = artifact.get('artifact_id', None)
+        if artifact.__class__.__name__ == "ArtifactInstallProperties":
+            if len(artifact.parameters) > 0:
```

Comment:
You can simply the condition to `if artifacts.parameters:`. In Python, when a collection is empty its __truthiness__ is __False__.

Reply:
resolved